MRIOperators/src/FieldmapNFFTOp.jl

```diff
@@ -93,45 +97,48 @@ function FieldmapNFFTOp(shape::NTuple{D,Int64}, tr::Trajectory,
     idx[κ] = findall(x->x!=0.0, cparam.A_k[:,κ])
     plans[κ] = plan_nfft(nodes[:,idx[κ]], shape, m=3, σ=1.25, precompute = NFFT.POLYNOMIAL)
   end
-
-  d = [zeros(ComplexF64, length(idx[κ])) for κ=1:K ]
-  p = [zeros(Complex{T}, shape) for κ=1:K]
-  x_tmp = zeros(Complex{T}, ncol)
-  y_tmp = zeros(Complex{T}, nrow)
   
-  circTraj = isCircular(tr)
+  d = [zeros(Complex{T}, length(idx[κ])) for κ=1:K ]
+  p = [zeros(Complex{T}, shape) for κ=1:K]
 
-  mul! = (res,x) -> produ!(res,x,x_tmp,shape,plans,idx,cparam,circTraj,d,p)
-  ctmul! = (res,y) -> ctprodu!(res,y,y_tmp,shape,plans,idx,cparam,circTraj,d,p)
+  circTraj = isCircular(tr)
 
   return FieldmapNFFTOp{T,Nothing,Function,D}(nrow, ncol, false, false
-            , mul!
+            , (res,x) -> produ!(res,x,x_tmp,shape,plans,idx,cparam,circTraj,d,p)
             , nothing
-            , ctmul!, 0, 0, 0, false, false, false, ComplexF64[], ComplexF64[]
+            , (res,y) -> ctprodu!(res,y,y_tmp,shape,plans,idx,cparam,circTraj,d,p), 0, 0, 0, false, false, false, Complex{T}[], Complex{T}[]
             , plans, idx, circTraj, shape, cparam)
 end
 
+function Base.copy(cparam::InhomogeneityData{T}) where T
+
+  return cparam
+
+end
+
 function Base.copy(S::FieldmapNFFTOp{T,Nothing,Function,D}) where {T,D}
+
+  shape = S.shape
+
   K=length(S.plans)
   plans = [copy(S.plans[i]) for i=1:K]
-  idx = deepcopy(S.idx)
+  idx = copy(S.idx)
 
-  d = [zeros(Complex{T}, length(idx[κ])) for κ=1:K ]
-  p = [zeros(Complex{T}, S.shape) for κ=1:K]
   x_tmp = zeros(Complex{T}, S.ncol)
   y_tmp = zeros(Complex{T}, S.nrow)
 
-  cparam = deepcopy(S.cparam)
+  cparam = copy(S.cparam)
+  d = [zeros(Complex{T}, length(idx[κ])) for κ=1:K]
+  p = [zeros(Complex{T}, shape) for κ=1:K]
 
-  mul! = (res,x) -> produ!(res,x,x_tmp,S.shape,plans,idx,cparam,S.circTraj,d,p)
-  ctmul! = (res,y) -> ctprodu!(res,y,y_tmp,S.shape,plans,idx,cparam,S.circTraj,d,p)
```

Comment:
personally i prefer having definitions like this separated like they were here
so that one can easily inspect them with @show etc.
rather than embedded in a long function call...
was this change needed to avoid the memory issue?

Reply:
@JeffFessler separating them seems to have no impact on the memory issue, I will separate them and call them mul! and ctmul! as before

Reply:
^pushed the change